client/my-sites/stats/stats-module-utm/utm-export-button.jsx

```diff
@@ -0,0 +1,69 @@
+import { Button, Gridicon } from '@automattic/components';
+import { saveAs } from 'browser-filesaver';
+import { useTranslate } from 'i18n-calypso';
+
+function UTMExportButton( { data } ) {
+	// Button set up.
+	const translate = useTranslate();
+	const buttonLabel = translate( 'Download data as CSV' );
+	const shouldDisableExport = data && data.length !== 0 ? false : true;
+
+	// Turns the working data into a flattened array of objects.
+	// Preserves the original data but updates the label for export.
+	const flattenDataForExport = ( originalData ) => {
+		const newData = [];
+		// Map the array of objects.
+		originalData.forEach( ( row ) => {
+			// Wrap label in quotes and push values.
+			let newLabel = `"${ row.label }"`;
+			newData.push( { label: newLabel, value: row.value } );
+			// Flatten children if present.
+			const children = row?.children;
+			if ( children ) {
+				const newChildren = children.map( ( child ) => {
+					newLabel = `"${ row.label } > ${ child.label }"`;
+					return { ...child, label: newLabel };
+				} );
+				newData.push( ...newChildren );
+			}
+		} );
+		return newData;
+	};
+
+	// Turns the flat array into a CSV string.
+	const prepareDataForDownload = ( flatData ) => {
+		const csvData = flatData
+			.map( ( row ) => {
+				return `${ row.label },${ row.value }`;
+			} )
+			.join( '\n' );
+
+		return csvData;
+	};
+
+	const initiateDownload = ( event ) => {
+		event.preventDefault();
+
+		// TODO: Provide a better default file name.
+		const fileName = 'my-data.csv';
```

Comment:
We could copy and tweak this to start with:

https://github.com/Automattic/wp-calypso/blob/ae401a84d6e9fe28cefc9dc4fd3045f4891d1746/client/my-sites/stats/stats-download-csv/index.jsx#L33

Reply:
That looks like a good starting point! Thanks.

Reply:
Addressed in #88708. Thanks again, Jasper!